### ISSUE TITLE
New audio in project to be selected automatically

### ISSUE THREAD

**Issue by SteveDaulton** [audacity/audacity#1725]
**Is your feature request related to a problem? Please describe.**
Audacity is inconsistent in selecting (or not) newly added audio.
1. When running a macro on "files", the imported track is automatically selected.
2. When manually importing a file, the imported track is not selected.
3. When generating audio, the new audio is selected.
4. When recording audio, the new audio is not selected

**Describe the solution you'd like**
New audio in a project to be automatically selected (consistently).

**Describe alternatives you've considered**
Living with the current mishmash.

**Additional context**
We receive complaints on the forum about effects throwing an error when the user has forgotten to select the track.

Previously (years ago), Audacity had a default behaviour of selecting the entire project if an effect was run without a selection, but this was highly problematic as in many cases the user did not want to apply the action to the entire project. In the worst case, a user might panic because they accidentally pressed the Del key and the entire project "disappeared".

I believe the main source of complaints comes from the situation where new audio has been added (typically from importing) and the user intuitively expects to be able to immediately apply an effect to it, but they can't because it is not selected.

**Comment by petersampsonaudacity** [comment]
I do not support this issue as it stands.

1) Recording
>4. When recording audio, the new audio is not selected

This then becomes an inconsistency

2) Mix &Render
for consistency, Mix&Render to New Track would need to atuto-select the new track after the mix.
_Note that this is not the case currently, rather the selection remains with the tracks selected for the mix._

Accordingly my counter-proposal is that we **remove** the auto-selection of 
>3. When generating audio, the new audio is selected.

This is the least work - and provides great consistency.

**Comment by petersampsonaudacity** [comment]
@SteveDaulton 's point1
>1. When running a macro on "files", the imported track is automatically selected.

Kind of makes sense as otherwise the user has to add a Select command to the Macro - but perhaps that is no bad thing - and it's not exactly arduous or tricky to do when you are writing a Macro.

So for absolute consistency in the handling of new audio I would also suggest that Audacity does **NOT** auto-select  imported audio when running a Macro on files.

**Comment by SteveDaulton** [comment]
> 1. Recording
> This then becomes an inconsistency

No, I'm suggesting that newly added audio should always (consistently) be selected. It should be selected whether it was added by importing, or generating, or recording, or any other way.

> 2. Mix &Render
Currently we have an inconsistency here too.
After "Mix and Render", the new track is selected, but after "Mix and Render to New  Track" it isn't.

>  it's not exactly arduous or tricky to do when you are writing a Macro.
No it's not particularly arduous or tricky, but it is unnecessary and would not be at all intuitive. Consider for example:
1. Import File
2. Normalize
3. Export as WAV
That all  seems very straightforward and obvious, but if the imported file was not automatically selected, then the macro would fail, leaving a very confused user scratching their head.

Surely, if that works in a macro (which it currently does), then it should also work manually. I see no good reason to force the user to add an additional step.

**Comment by petersampsonaudacity** [comment]
> No it's not particularly arduous or tricky, but it is unnecessary and would not be at all intuitive. Consider for example:
> 
>     1. Import File
>     2. Normalize
>     3. Export as WAV

But normally such a Macro would be operating on an audio file - so why would you need step 1 ?

A good example to observe is the shipped Macro "MP3 Conversion"
>1. Normalize:ApplyGain="1" PeakLevel="-1" RemoveDcOffset="1" StereoIndependent="0"
>2. ExportMP3:

**Comment by SteveDaulton** [comment]
> A good example to observe is the shipped Macro "MP3 Conversion"
> 
> > 1. Normalize:ApplyGain="1" PeakLevel="-1" RemoveDcOffset="1" StereoIndependent="0"
> > 2. ExportMP3:

Importing the file is implicit because batch processing is passed a list of files to import.

I am still unable to think of a use case where selecting the new audio would be a bad thing. Please give some examples.

**Comment by LWinterberg** [comment]
I feel like the solution here would be to

1. always apply effects to the *track*, as opposed to the entire project, when nothing is selected, and
2. not select anything upon creating audio
